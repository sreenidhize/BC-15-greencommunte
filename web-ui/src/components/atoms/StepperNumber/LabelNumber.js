import { Avatar, makeStyles } from "@material-ui/core";
import React from "react";

const useStyles = makeStyles((theme) => ({
  unvisited: {
    color: "#5f7381",
    backgroundColor: "#e3f3f6",
    marginRight: "15px",
    fontSize: "18px",
    fontFamily: theme.typography.fontFamily,
  },
  visited: {
    color: "white",
    backgroundColor: theme.palette.primary.main,
    marginRight: "15px",
    fontSize: "18px",
    fontFamily: theme.typography.fontFamily,
  },
}));

const LabelNumber = (props) => {
  const classes = useStyles();
  return (
    <Avatar
      className={props.isVisited ? classes.visited : classes.unvisited}
      size="small"
      data-testid="label-number"
    >
      {props.number}
    </Avatar>
  );
};

export default LabelNumber;
