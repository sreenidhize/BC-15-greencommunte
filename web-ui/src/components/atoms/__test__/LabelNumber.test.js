import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import LabelNumber from "../StepperNumber/LabelNumber";


describe("LabelNumber test", () => {
    it("Matches the snapshot", () => {  
      const { container } = render(
        <LabelNumber isVisited={false} number="1"/>
      );
      expect(container).toMatchSnapshot();
    });
  
    test("Test LabelNumber render", () => {
      const { getByTestId } = render(<LabelNumber isVisited={false} number="1"  />);
      expect(getByTestId("label-number")).toBeInTheDocument();
    });
  
    test("Test filled LabelNumber render", () => {
      const { getByTestId } = render(<LabelNumber isVisited={true} number="2" />);
      expect(getByTestId("label-number")).toBeInTheDocument();
    });
  });