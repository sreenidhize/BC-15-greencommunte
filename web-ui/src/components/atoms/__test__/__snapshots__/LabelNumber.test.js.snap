// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelNumber test Matches the snapshot 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circle makeStyles-unvisited-1 MuiAvatar-colorDefault"
    data-testid="label-number"
  >
    1
  </div>
</div>
`;
